lib/rubocop/cop/rspec/leaked_const.rb

```diff
@@ -0,0 +1,61 @@
+module RuboCop
+  module Cop
+    module RSpec
+      # Check that constants are not created in tests without cleanup.
+      #
+      # Prefer stub_const and anonymous classes.
+      #
+      # @example
+      #   # bad
+      #   describe MyClass do
+      #     class Foo
+      #       BAR = 1
+      #     end
+      #   end
+      #
+      #   # good
+      #   describe MyClass do
+      #     before do
+      #       stub_const('Foo::BAR', 1)
+      #     end
+      #   end
+      class LeakedConst < Cop
+        MSG = 'Opening a class to define methods can pollute your tests. '\
+              'Instead, try using `stub_const` with an anonymized class.'.freeze
+
+        def_node_matcher :stub_const, <<-PATTERN
+          (begin (send nil :stub_const $_ ...) ...)
```

Comment:
Since we're checking for usages of stub_const on the class definition's parent, the begin is required to catch the before block.